pygen/pygen_src/isa/riscv_instr.py

```diff
@@ -0,0 +1,341 @@
+from collections import defaultdict
+from pygen_src.riscv_instr_gen_config import *
+from pygen_src.riscv_instr_pkg import *
+from pygen_src.isa import rv32i_instr
+import random
+from bitstring import BitArray, BitStream
+import logging
+from copy import copy
+class riscv_instr:
+    instr_registry = {}
+    def __init__(self): 
+        self.instr_names = []
+        self.instr_name = None
+        self.instr_group =  defaultdict(list) 
+        self.instr_category = defaultdict(list)
+        self.basic_instr = []
+        self.instr_template = {}
+        
+        self.exclude_reg = []
+        self.include_reg = []
+        
+        self.imm_len = 0
```

Comment:
I think `self.imm_type` is missing

Reply:
please also add `group`, `format`, and `category` fields

Reply:
Added and committed the changes.